pkg/urbit/vere/pier.c

```diff
@@ -1035,11 +1087,15 @@ _pier_boot_ready(u3_pier* pir_u)
   u3_lord* god_u = pir_u->god_u;
   u3_disk* log_u = pir_u->log_u;
 
-  c3_assert( c3y == god_u->liv_o );
-  c3_assert( c3y == log_u->liv_o );
   c3_assert( u3_psat_init == pir_u->sat_e );
-  c3_assert( c3n == pir_u->liv_o );
-  pir_u->liv_o = c3y;
+
+  if ( ( 0 == god_u) ||
+       ( 0 == log_u) ||
+       (c3y != god_u->liv_o) ||
+       (c3y != log_u->liv_o) )
+  {
+    return;
```

Comment:
Looks like this now returns, but it used to die. Why the change?

Reply:
I changed the calling convention -- the (two) callers used to verify the preconditions before calling, now the function itself is resilient to being called twice in either order (ie, disk initialized before serf and vice versa) and the call-sites are simpler.